common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -303,6 +303,101 @@ def max_score(self):
+            # Anonymous ID is required for Matlab, CodeResponse, and some custom problems that include
+            # '$anonymous_student_id' in their XML.
+            # For the purposes of this report, we don't need to support those use cases.
+            anonymous_student_id=None,
+            cache=None,
+            can_execute_unsafe_code=lambda: None,
+            get_python_lib_zip=lambda: None,
+            DEBUG=None,
+            filestore=self.runtime.resources_fs,
+            i18n=self.runtime.service(self, "i18n"),
+            node_path=None,
+            render_template=None,
+            seed=1,
+            STATIC_URL=None,
+            xqueue=None,
+            matlab_api_key=None,
+        )
+
+        for idx, user_state in enumerate(user_state_iterator):
+
+            if limit_responses and idx >= limit_responses:
+                break
+
+            if 'student_answers' not in user_state.state:
+                continue
+
+            lcp = LoncapaProblem(
+                problem_text=self.data,
+                id=self.location.html_id(),
+                capa_system=capa_system,
+                # We choose to run without a fully initialized CapaModule
+                capa_module=None,
+                state={
+                    'done': user_state.state.get('done'),
+                    'correct_map': user_state.state.get('correct_map'),
+                    'student_answers': user_state.state.get('student_answers'),
+                    'has_saved_answers': user_state.state.get('has_saved_answers'),
+                    'input_state': user_state.state.get('input_state'),
+                    'seed': user_state.state.get('seed'),
+                },
+                seed=user_state.state.get('seed'),
+                # extract_tree=False allows us to work without a fully initialized CapaModule
+                # We'll still be able to find particular data in the XML when we need it
+                extract_tree=False,
+            )
+
+            for answer_id, orig_answers in lcp.student_answers.items():
+                # Some types of problems have data in lcp.student_answers that isn't in lcp.problem_data.
+                # E.g. formulae do this to store the MathML version of the answer.
+                # We exclude these rows from the report because we only need the text-only answer.
+                if answer_id.endswith('_dynamath'):
+                    continue
+
+                # The rest of items are answers for which we have information
+                assert answer_id in lcp.problem_data
+
+                question_text = lcp.find_question_label(answer_id)
+                answer_text = lcp.find_answer_text(answer_id, current_answer_text=orig_answers)
+
+                yield (user_state.username, {
+                    "Answer ID": answer_id,
+                    "Question": question_text,
+                    "Answer": answer_text,
+                })
```

Comment:
Could you create a namedtuple for this? Perhaps StudentResponseData or something.

Reply:
Then the definition of the tuple would be very far away from where it's used. We would need to updated the PR too.
But maybe we should use more variable-like keys, like "answer_id', "question", "answer". Or not. The implementation document uses "Question", "Answer".

Reply:
I think ``Question`` and ``Answer`` are fine since in the end it's supposed to be human-readable. I guess it makes sense not to create a NamedTuple since there is no standard shape for the state either.

Reply:
I like `Question` and `Answer` as well, but please wrap these strings in a gettext call; `_` is available here too.